lib/tree.rb

```diff
@@ -16,47 +16,168 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  # Time Complexity: O(log(n)) - Assuming the tree is somewhat balanced, this could potentially be O(log(n)) because we may have to traverse the height of the tree to get to a leaf node. 
+  # Space Complexity: O(n) - for recursive solution, there is space in the form of the call stack
+  # Space Complexity: O(1) - for while loop
+  
+  def add(key, value = nil)
```

Comment:
👍 Both the space/time complexities are O(n) if the tree is unbalanced and O(log n) if the tree is balanced.   Good use of default parameters